core/modules/locale/locale.theme.inc

```diff
@@ -101,3 +101,54 @@ function theme_locale_date_format_form($variables) {
   return $output;
 }
 
+/**
+ * Returns themed HTML for the language switcher block.
+ *
+ * @param $variables
+ *   An associative array containing:
+ *   - links: (object) The output of language_negotiation_get_switch_links():
+ *     - links: An associativearray of language links to be rendered.
+ *     - provider: ???
+ *   - settings: An associative array with the display settings for the language
+ *     switcher block:
+ *     - language_list_display: either "vertical" or "inline".
+ *     - language_link_label_display: either "name" or "code".
+ *     - custom_delimiter: either completely omitted (falls back to "|"), or a
+ *     single character, to be used as a delimiter when the language links are
+ *     to be rendered inline. This is added to the list items as a
+ *     data-list-delimiter attribute, and can be used in CSS via a 'content'
+ *     property. For example:
+ *     @code
+ *     .language-switcher.links.inline li:after {
+ *       content: attr(data-list-delimiter);
+ *       padding: 0 0.6em;
+ *       color: gray;
+ *     }
+ *     @endcode
+ *
+ * @ingroup themeable
+ */
+function theme_locale_block($variables) {
```

Comment:
I don't think we should use the word "block" in this theme function, since it is not a block. This theme function could be called from anywhere, and it would produce a list of links. 

Can we use `theme_links__locale()` instead?